pkg/objstore/s3/s3.go

```diff
@@ -173,8 +183,32 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 	client.SetAppInfo(fmt.Sprintf("thanos-%s", component), fmt.Sprintf("%s (%s)", version.Version, runtime.Version()))
 
 	var sse encrypt.ServerSide
-	if config.SSEEncryption {
-		sse = encrypt.NewSSE()
+	if config.SSEConfig.Type != "" {
+		switch {
+		case config.SSEConfig.Type == "SSE-KMS":
```

Comment:
Why not 

```
switch config.SSEConfig.Type {
case SEEKMS: 
...
}
```
(let's put those strings in constant)

Reply:
Done :D